test/extended/setup.sh

```diff
@@ -11,17 +11,25 @@ function os::test::extended::focus () {
 	if [[ -n "${FOCUS:-}" ]]; then
 		exitstatus=0
 
+		local excludesFile="${ARTIFACT_DIR}/parallel_test_names.txt"
+
 		# first run anything that isn't explicitly declared [Serial], and matches the $FOCUS, in a parallel mode.
 		os::log::info "Running parallel tests N=${PARALLEL_NODES:-<default>} with focus ${FOCUS}"
-		TEST_REPORT_FILE_NAME=focus_parallel TEST_PARALLEL="${PARALLEL_NODES:-5}" os::test::extended::run -- -ginkgo.skip "\[Serial\]" -test.timeout 6h ${TEST_EXTENDED_ARGS-} || exitstatus=$?
+		TEST_REPORT_FILE_NAME=focus_parallel TEST_PARALLEL="${PARALLEL_NODES:-5}" \
+			os::test::extended::run -- \
+				-print-test-names-to "${excludesFile}" \
+				-ginkgo.skip "\[Serial\]" \
+				-test.timeout 6h ${TEST_EXTENDED_ARGS-} || exitstatus=$?
 
 		# Then run everything that requires serial and matches the $FOCUS, serially.
 		# there is bit of overlap here because not all serial tests declare [Serial], so they might have run in the
 		# parallel section above.  Hopefully your focus was precise enough to exclude them, and we should be adding
 		# the [Serial] tag to them as needed.
 		os::log::info ""
 		os::log::info "Running serial tests with focus ${FOCUS}"
-		TEST_REPORT_FILE_NAME=focus_serial os::test::extended::run -- -suite "serial.conformance.openshift.io" -test.timeout 6h ${TEST_EXTENDED_ARGS-} || exitstatus=$?
+		TEST_REPORT_FILE_NAME=focus_serial os::test::extended::run -- \
+			-excludes-from-file "${excludesFile}" \
+			-test.timeout 6h ${TEST_EXTENDED_ARGS-} || exitstatus=$?
```

Comment:
this still needs to set the serial suite, otherwise you're going to run everything that wasn't run parallel, which will include tests the user wasn't trying to FOCUS on.

Reply:
Currently, `-suite` argument overrides the focus entirely.

With focus applied to both parallel and serial tests, where the former excluded `\[Serial\]` tests, we only need to run those excluded from the former run (while still keeping the focus).

Reply:
@bparees as I can see "serial.conformance.openshift.io" shouldn't be used there, pruning tests are not labeled as conformance tests.